mqtt/mqtt-bridge/src/messages.rs

```diff
@@ -81,20 +83,53 @@ impl<S> StoreMqttEventHandler<S> {
 
     fn transform(&self, topic_name: &str) -> Option<String> {
         self.topic_mappers.values().find_map(|mapper| {
-            mapper
-                .topic_settings
-                .in_prefix()
-                // maps if local does not have a value it uses the topic that was received,
-                // else it checks that the received topic starts with local prefix and removes the local prefix
-                .map_or(Some(topic_name), |local_prefix| {
-                    topic_name.strip_prefix(format!("{}/", local_prefix).as_str())
-                })
-                // match topic without local prefix with the topic filter pattern
-                .filter(|stripped_topic| mapper.topic_filter.matches(stripped_topic))
-                .map(|stripped_topic| match mapper.topic_settings.out_prefix() {
-                    Some(remote_prefix) => format!("{}/{}", remote_prefix, stripped_topic),
-                    None => stripped_topic.to_string(),
-                })
+            if mapper.topic_filter.matches(topic_name) {
+                mapper
+                    .topic_settings
+                    .in_prefix()
+                    // maps if local does not have a value it uses the topic that was received,
+                    // else it checks that the received topic starts with local prefix and removes the local prefix
+                    .map_or(Some(topic_name), |local_prefix| {
```

Comment:
Nit: should we call this in_prefix and out_prefix? The local_prefix here can be actually remote prefix for remote subscriptions, right?